MapboxNavigation/InstructionPresenter.swift

```diff
@@ -1,16 +1,27 @@
 import UIKit
 import MapboxDirections
 
+protocol InstructionPresenterDataSource: class {
+    var availableBounds: (() -> CGRect)! { get }
```

Comment:
Still no solution to eliminate this closure? I guess we can flag it as tail work... now that we have this protocol it feels like a leaky abstraction (and maybe it isn't needed?). Just dropping a note here for posterity; I don't think it should block merging this PR since all we've done here is introduce a protocol to better clarify presenter's data dependencies.

Reply:
Curious about an alternative solution. 🤔